hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/DFSUtil.java

```diff
@@ -647,6 +634,58 @@ public static String addKeySuffixes(String key, String... suffixes) {
       getNNLifelineRpcAddressesForCluster(Configuration conf)
       throws IOException {
 
+    Collection<String> parentNameServices = getParentNameServices(conf);
+
+    return getAddressesForNsIds(conf, parentNameServices, null,
+        DFS_NAMENODE_LIFELINE_RPC_ADDRESS_KEY);
+  }
+
+  //
+  /**
+   * Returns the configured address for all NameNodes in the cluster.
+   * This is similar with DFSUtilClient.getAddressesForNsIds()
+   * but can access DFSConfigKeys.
+   *
+   * @param conf configuration
+   * @param defaultAddress default address to return in case key is not found.
+   * @param keys Set of keys to look for in the order of preference
+   *
+   * @return a map(nameserviceId to map(namenodeId to InetSocketAddress))
+   */
+  static Map<String, Map<String, InetSocketAddress>> getAddressesForNsIds(
```

Comment:
Why adding this function here instead of changing the function in the DFSUtilClient.getAddressesForNsIds ?

Reply:
The thing is DFSUtilClient does not have access to DFSConfigKeys as it is in hadoop-hdfs-client module... So I pulled it out here to read the DFSConfigKeys fields.

Please suggest if you find any better ways to do this.

Reply:
Yeah that sounds better, let me try it out